src/firefly/js/charts/ui/HistogramOptions.jsx

```diff
@@ -81,70 +81,64 @@ var columnNameReducer= (colValStats) => {
+                                       const dataMin = colValStats[i].min;
+                                       const dataMax = colValStats[i].max;
+                                       var binWidth = ((dataMax - dataMin) / numBins).toFixed(6);
+                                       inFields = updateSet(inFields, ['minCutoff', 'value'], `${dataMin}`);
+                                       inFields = updateSet(inFields, ['maxCutoff', 'value'], `${dataMax}`);
+                                       inFields = updateSet(inFields, ['binWidth', 'value'], `${binWidth}`);
+                                       break;
+                                   }
+                               }
+                           }
+                           else {
+                               inFields = updateSet(inFields, ['minCutoff', 'value'], undefined);
+                               inFields = updateSet(inFields, ['maxCutoff', 'value'], undefined);
+                               inFields = updateSet(inFields, ['binWidth', 'value'], undefined);
+                           }
+                         }
 
+                    }
                     break;
-                case 'numBins':
-                case 'minCutoff':
-                case 'maxCutoff':
-                    const numBins = get(inFields, ['numBins','value'], 50);
-                    const cName   = get(inFields, ['columnOrExpr','value'], undefined);
-                    if (!cName) break;
-                    for (let i=0; i<colValStats.length; i++){
-                        if (cName=== colValStats[i].name){
-                            const dataMin = get(inFields,['minCutoff','value'], colValStats[i].min);
-                            const dataMax = get(inFields,['maxCutoff','value'],  colValStats[i].max);
-                            var  binWidth = ((dataMax - dataMin)/numBins).toFixed(6);
-
-                            inFields = updateSet(inFields, ['minCutoff', 'value'],`${dataMin}`);
-                            inFields = updateSet(inFields, ['maxCutoff', 'value'], `${dataMax}`);
-                           // inFields = updateSet(inFields, ['binWidth', 'value'], `${binWidth}`);
-                            if (isFinite(parseFloat(binWidth)) ){
-                                inFields = updateSet(inFields, ['binWidth', 'value'], `${binWidth}`);
-                            }
-                            else {
-                                inFields = updateSet(inFields, ['binWidth', 'value'], '');
-                            }
-                            break;
-
-                        }
+               //When numBins changes, update binWidth
+                case 'numBins':  {
+                         const numBins = get(inFields, ['numBins', 'value'], 50);
+                         const min = get(inFields, ['minCutoff', 'value'], '');
+                         const max = get(inFields, ['maxCutoff', 'value'], '');
```

Comment:
I don't expect a value unless the column name is a single column.  For expression, those two fields are not empty.